sklearn/utils/__init__.py

```diff
@@ -176,10 +176,14 @@ def _array_indexing(array, key, key_dtype, axis):
             key = np.asarray(key)
     if isinstance(key, tuple):
         key = list(key)
+    if inverse:
+        mask = np.ones(array.shape[0] if axis == 0 else array.shape[1], bool)
+        mask[key] = False
+        key = mask
     return array[key] if axis == 0 else array[:, key]
 
 
-def _pandas_indexing(X, key, key_dtype, axis):
+def _pandas_indexing(X, key, key_dtype, axis, inverse):
```

Comment:
I'd definitely need a careful review on this one. Also @glemaitre ?

Reply:
Maybe add unit tests to ease the review